packages/paste-core/components/combobox/src/multiselect/MultiselectCombobox.tsx

```diff
@@ -0,0 +1,314 @@
+              } else {
+                addSelectedItem(newSelectedItem);
+              }
+            }
+            break;
+          default:
+            break;
+        }
+      },
+    });
+
+    const size = React.useMemo(() => items.length, [items]);
+    // Virtualizer for long lists that don't use `groupItemsBy`
+    const rowVirtualizer = useVirtual({
+      size,
+      parentRef,
+      overscan: 8,
+    });
+
+    // Fixes issue where dynamic height items would not recompute on window resize and list content wrapping
+    React.useEffect(() => {
+      rowVirtualizer.measure();
+    }, [width]);
+
+    // Use ref to focus the current selected item when the list is opened
+    // https://tkdodo.eu/blog/avoiding-use-effect-with-callback-refs
+    const scrollToIndexRef = React.useCallback(
+      (node) => {
+        // FEAT: Scroll to the last selected item when the list is opened
+        // When the list is opened, highlightedIndex is set to -1, so that's how we check if the list is opened
+        // We also have a fix in the `stateReducer` to handle setting highlightedIndex to -2 when the
+        // menu is blurred instead of being set to -1, which confuses this check.
+        if (node && highlightedIndex === -1) {
+          const lastSelectedItem = selectedItems[selectedItems.length - 1];
+          rowVirtualizer.scrollToIndex(items.indexOf(lastSelectedItem));
+        }
+
+        // FIX: Looping fix, scroll to the beginning of the list when the first item is highlighted
+        if (node && highlightedIndex === 0) {
+          rowVirtualizer.scrollToIndex(0);
+        }
+        // FIX: Looping fix, scroll to the end of the list when the last item is highlighted
+        if (node && highlightedIndex === size - 1) {
+          rowVirtualizer.scrollToIndex(size - 1);
+        }
+      },
+      [items, selectedItems, highlightedIndex]
+    );
+
+    // FIX: a11y issue where `aria-expanded` isn't being set until the dropdown opens the very first time
+    const comboboxProps = getComboboxProps({
+      disabled,
+      role: 'combobox',
+    });
+    const ariaExpanded = comboboxProps['aria-expanded'] || 'false';
+
+    return (
+      <Box position="relative" element={`${element}_WRAPPER`}>
+        <Label disabled={disabled} required={required} variant={variant} {...getLabelProps()}>
+          {labelText}
+        </Label>
+        <InputBox
+          disabled={disabled}
+          hasError={hasError}
+          element={element}
+          insertBefore={insertBefore}
+          insertAfter={insertAfter}
+          variant={variant}
+        >
+          <Box
+            {...getToggleButtonProps({disabled})}
+            {...comboboxProps}
+            aria-expanded={ariaExpanded}
+            position="relative"
+            width="100%"
+            paddingY="space30"
+            paddingX="space40"
+          >
+            <FormPillGroup
+              {...pillState}
+              aria-label="Products:"
```

Comment:
issue: hard coded string.

Reply:
just came across a problem with our plan to override `i18nKeyboardControls` so it's blank, and place that hidden keyboard controls text _after_ the combobox element. The `aria-decribedby` prop that you would also need to override to point this to the new hidden keyboard controls, isn't overridable in the form pill group.

Reply:
Good catch. Fixed.